backend/src/main/java/com/bakdata/conquery/models/index/MapIndexService.java

```diff
@@ -72,18 +96,41 @@ public Map<String, String> load(@NotNull Key key) throws Exception {
 		}
 	});
 
+	public void evictCache() {
+		synchronized (requestedMappings) {
+			// Invalidate all mapping requests by cancelling them
+			final Iterator<CompletableFuture<Map<String, String>>> iterator = requestedMappings.iterator();
+
+			while (iterator.hasNext()) {
+				final CompletableFuture<Map<String, String>> next = iterator.next();
+
+				next.obtrudeException(new CancellationException("The mapping for this future was evicted"));
+				iterator.remove();
```

Comment:
Verstehe ich das richtig, dass du hiermit laufende Anfragen Canceln willst, und das unten für Retry Logik verwendest? 

Mir kommt das noch sehr ambitioniert vor, hier ganz offenes async zu haben, wenn ich ein Konzept Update und danach ein UpdateMatchingStats habe, kann ich ja ggf echt in eine Situation kommen wo die Erzeugung noch läuft und es mir dann timeouted.

Reply:
Siehst du ein Problem wenn während eines Konzept-Updates evicted wird? 
Das mit dem UpdateMatchingStats verstehe ich gerade nicht. Es hat gerade keinen Einfluss auf das Mapping.

Reply:
Zu dem Zeitpunkt dachte ich noch, dass du schon in den Filtern die Referenzen hast. Wäre aber trotzdem was worüber du Gedanken machen musst, sonst schießt du ggf ein schlecht getimtes MatchingStats ab